### PR TITLE
add esmf810 updates

### PR DESCRIPTION
**Pull request by junwang-noaa**
Add GFSv16 nems updates for performance tuning

Associated PRs:

[fv3atm PR#160](https://github.com/NOAA-EMC/fv3atm/pull/160)
[ufs-weather-model PR #188](https://github.com/ufs-community/ufs-weather-model/pull/188)

this [NEMS #PR76](https://github.com/NOAA-EMC/NEMS/pull/76) needs to be committed before ufs-weather-model PR #188.

### CODE REVIEW

File: src/incmake/component_WW3.mk [hunk 1]
```diff
@@ -37,7 +37,7 @@ WW3_ALL_OPTS= \
 $(ww3_mk): configure
 	+$(MODULE_LOGIC) ; set -x ; cd $(WW3_SRCDIR)/esmf       ; \
 	export $(WW3_ALL_OPTS)                                  ; \
-	exec $(MAKE) -j 1 WW3_COMP="$(WW3_CONFOPT)" ww3_nems
+	exec $(MAKE) -j 1 WW3_COMP="$(WW3_CONFOPT)" ww3_nemslibonly
```
Comment (JessicaMeixner-NOAA) [inline src/incmake/component_WW3.mk:40]: While this change will make WW3 build faster and perhaps circumvent other build issues, currently there is not another work around for building the other WW3 jobs in the workflow. Until this is there, I don't think this is a good idea.

**Comment by junwang-noaa** [comment]
The change is not to make ww3 run faster, without it the ufs-weather can't
even compile with ww3 using gnu make. Will any person from WW3 take a look
at it? We can use cmake build though.

**Comment by SMoorthi-emc** [comment]
Jun,
Is this compilation not using “compile.sh”?
I wonder why I can compile without this change under s2s.
Moorthi

**Comment by junwang-noaa** [comment]
Moorthi, the new NCEPLIBS build is only in ufs-weather and not used by s2s
yet. But we'd like to use the same NCEPLIBS in both models in the future.
The issue in current ufs-weather model is that ww3 needs to build both ww3
lib required for the model and the ww3 pre- and post- procoessing
executable, and there is some issue with these executable to use the new
NCEPLIBS.

File: src/module_EARTH_GRID_COMP.F90 [hunk 2]
```diff
@@ -256,10 +256,12 @@ SUBROUTINE EARTH_REGISTER(EARTH_GRID_COMP,RC_REG)
         specLabel=Driver_label_SetRunClock, specRoutine=NUOPC_NoOp, rc=RC_REG)
       ESMF_ERR_RETURN(RC,RC_REG)
       
+#if 0
```
Comment (DusanJovic-NOAA) [inline src/module_EARTH_GRID_COMP.F90:259]: If this call is not needed anymore, delete it.

**Comment by junwang-noaa** [comment]
There was a discussion with Gerhard on this, I'd like to have a version
matching GFS.v16, then delete it in a later commit. I will create an issue
in nems so that we won't forget.

File: src/incmake/component_WW3.mk (hunk 1 above)
Comment (junwang-noaa) [reply src/incmake/component_WW3.mk:40]: After discussion, Ali/Arun will work on a wrapper script to compile the ww3 pre-/post- processing executable. So ww3 will not dependent on model to create those executable.

File: src/module_EARTH_GRID_COMP.F90 (hunk 2 above)
Comment (junwang-noaa) [reply src/module_EARTH_GRID_COMP.F90:259]: We will keep same version as current GFSv16 production branch and remove it in a later commit (possibly s2s lib updates).

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**